database/migrations/create_chat_tables.php

```diff
@@ -28,145 +27,66 @@ public function __construct()
-                $table->boolean('flagged')->default(false);
-                $table->timestamps();
-                $table->softDeletes();
-
-                $table->index(['user_id', 'message_id']);
-
-                $table->foreign('message_id')
-                    ->references('id')->on('mc_messages')
-                    ->onDelete('cascade');
-
-                $table->foreign('conversation_id')
-                    ->references('id')->on('mc_conversations')
-                    ->onDelete('cascade');
-
-                $table->foreign('user_id')
-                    ->references($this->userModelPrimaryKey)
-                    ->on($this->userModelTable)
-                    ->onDelete('cascade');
-            });
-        }
+        Schema::create('mc_conversations', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->boolean('private')->default(true);
+            $table->text('data')->nullable();
+            $table->timestamps();
+        });
+
+        Schema::create('mc_messages', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->text('body');
+            $table->bigInteger('conversation_id')->unsigned();
+            $table->bigInteger('user_id')->unsigned();
+            $table->string('type')->default('text');
+            $table->timestamps();
+
+            $table->foreign('user_id')
+                ->references($this->userModelPrimaryKey)
+                ->on($this->userModelTable)
+                ->onDelete('cascade');
+
+            $table->foreign('conversation_id')
+                ->references('id')
+                ->on('mc_conversations')
+                ->onDelete('cascade');
+        });
+
+        Schema::create('mc_conversation_user', function (Blueprint $table) {
+            $table->bigInteger('conversation_id')->unsigned();
+            $table->bigInteger('messageable_id')->unsigned();
+            $table->string('messageable_type');
+            $table->primary(['conversation_id', 'messageable_id', 'messageable_type']);
+            $table->timestamps();
+
+            $table->foreign('conversation_id')
+                ->references('id')->on('mc_conversations')
+                ->onDelete('cascade');
+        });
+
+        Schema::create('mc_message_notification', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->bigInteger('message_id')->unsigned();
+            $table->bigInteger('conversation_id')->unsigned();
+            $table->bigInteger('messageable_id')->unsigned();
+            $table->string('messageable_type');
+            $table->boolean('is_seen')->default(false);
+            $table->boolean('is_sender')->default(false);
+            $table->boolean('flagged')->default(false);
+            $table->timestamps();
+            $table->softDeletes();
+
+            $table->index(['participation_id', 'message_id']);
```

Comment:
I got an error that says `Key column 'participation_id' doesn't exist in table` 🤔

Reply:
Oh yeah, i need to add that column to reference the participants table which is `mc_conversation_user` table

Reply:
This https://github.com/musonza/chat/pull/163/commits/1a19e4fa4a3f615ea1cf330c663e4c4627aaa4fa should address this. Update your `dev-multiple-model-recipients` branch. There is still some issues that may need to be fixed and i will address as soon as get time. Otherwise feel free to make PRs against that branch